packages/@n8n/task-runner/src/js-task-runner/result-validation.ts

```diff
@@ -9,7 +9,7 @@ export const REQUIRED_N8N_ITEM_KEYS = new Set(['json', 'binary', 'pairedItem', '
 function validateTopLevelKeys(item: INodeExecutionData, itemIndex: number) {
 	for (const key in item) {
 		if (Object.prototype.hasOwnProperty.call(item, key)) {
-			if (REQUIRED_N8N_ITEM_KEYS.has(key)) return;
+			if (REQUIRED_N8N_ITEM_KEYS.has(key)) continue;
```

Comment:
Lucky we caught this early!

1. I notice that there is some duplication between `./packages/@n8n/task-runner/src/js-task-runner/result-validation.ts` and `./packages/@n8n/task-runner/src/js-task-runner/result-validation.ts`. The versions risk drifting apart as validation is updated in the Code node. Should this validation exist only once in `n8n-core` and be imported into the runner and into the Code node sandbox?
2. I also notice the runner and the Code node sandbox do not share _all_ validation, e.g. `Access the properties of an item under .json, e.g. item.json` and `In the returned data, every key named 'json' must point to an object.` are duplicated, but `Please return an array of objects, one for each item you would like to output` and `Please return an object representing the output item. ('${executionResult}' was returned instead.)` exist only in the Code node sandbox. Is this an issue also?

Reply:
💯! Very good questions!
1. I assume you mean between task runner and the Code Node Sandbox? The Sandbox should eventually go away entirely, so that duplication will be solved by that.
2. I couldn't find any mismatches between the two. E.g. the `Please return an array of objects` is [in Sandbox](https://github.com/n8n-io/n8n/blob/cat-363-adding-a-property-to-the-item-top-level-does-not-throw/packages/nodes-base/nodes/Code/Sandbox.ts#L67-L73) and also [in the runner](https://github.com/n8n-io/n8n/blob/cat-363-adding-a-property-to-the-item-top-level-does-not-throw/packages/@n8n/task-runner/src/js-task-runner/result-validation.ts#L48-L51). Maybe you missed it because the code doesn't match 1:1 (the Sandbox uses some interpolation)